homeassistant/util/json.py

```diff
@@ -61,17 +62,16 @@ def save_json(
         raise SerializationError(msg) from error
 
     tmp_filename = ""
-    tmp_path = os.path.split(filename)[0]
     try:
         # Modern versions of Python tempfile create this file with mode 0o600
         with tempfile.NamedTemporaryFile(
-            mode="w", encoding="utf-8", dir=tmp_path, delete=False
+            mode="w", encoding="utf-8", delete=False
```

Comment:
Can we be 100% certain that the user running home assistant on the host has access to it?
If we can not we should not do this, as this will break the entire setup.
When it writes to the same dir as the target file, we know that this is the configuration directory of home assistant and that we have access to it.

I have seen users get premission issues with access to these default temp directories both on Windows and on a venv installation.

Reply:
I agree, if you have seen such case than we cannot risk. I will propose other solution.